client/my-sites/plans/jetpack-plans/product-store/types.ts

```diff
@@ -42,11 +42,15 @@ export interface ViewFilterProps {
 }
 
 export type ProductsListProps = ProductStoreBaseProps &
-	Omit< ProductStoreProps, 'urlQueryArgs' | 'header' >;
+	Omit< ProductStoreProps, 'urlQueryArgs' | 'header' > & {
+		clickMoreHandlerFactory: ( item: SelectorProduct ) => VoidFunction | undefined;
```

Comment:
We can name it something generic and an event handler starting with `on`.

`undefined` can also be removed after my [previous suggestion](https://github.com/Automattic/wp-calypso/pull/67389/files#r963305939).

```suggestion
		onClickMoreInfo: ( item: SelectorProduct ) => VoidFunction;
```

Reply:
Hi @manzoorwanijk , the one I pass here actually returns a VoidFunction that is the actual event handler. hence the factory at the name.

we can maybe use `onClickMoreInfoFactory` or something that emphasizes it is not the actual handler but a factory for handler. wdyt?

Reply:
That sounds good to me.